crates/biome_js_analyze/src/lint/correctness/no_unused_imports.rs

```diff
@@ -88,8 +98,160 @@ declare_lint_rule! {
     }
 }
 
+/// Model for tracking any types referenced in JsDoc or TSDoc comments (like {@link TypeName} or @param {TypeName})
+#[derive(Clone, Default)]
+struct JsDocTypeModel(FxHashSet<String>);
+
+impl Deref for JsDocTypeModel {
+    type Target = FxHashSet<String>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for JsDocTypeModel {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+#[derive(Default)]
+struct JsDocTypeCollectorVisitior {
+    jsdoc_types: JsDocTypeModel,
+}
+
+declare_node_union! {
+    pub AnyJsWithTypeReferencingJsDoc = AnyJsDeclaration | AnyJsClassMember | AnyTsTypeMember | TsEnumMember
+}
+
+impl Visitor for JsDocTypeCollectorVisitior {
+    type Language = JsLanguage;
+    fn visit(
+        &mut self,
+        event: &WalkEvent<SyntaxNode<Self::Language>>,
+        _ctx: VisitorContext<Self::Language>,
+    ) {
+        match event {
+            WalkEvent::Enter(node) => {
+                if AnyJsWithTypeReferencingJsDoc::can_cast(node.kind()) {
+                    load_jsdoc_types_from_node(&mut self.jsdoc_types, &node);
+                }
+            }
+            WalkEvent::Leave(_) => {}
+        }
+    }
+
+    fn finish(self: Box<Self>, ctx: VisitorFinishContext<JsLanguage>) {
+        ctx.services.insert_service(self.jsdoc_types);
+    }
+}
+
+fn load_jsdoc_types_from_node(model: &mut JsDocTypeModel, node: &SyntaxNode<JsLanguage>) {
+    if let Ok(comment) = JsdocComment::try_from(node) {
```

Comment:
Do the regular expressions need to be able to work across newlines? If not, maybe you can even use `JsdocComment::text_is_jsdoc_comment()` only and avoid instantiating the `JsdocComment` itself. The regular expressions may just work fine on the raw string, so you would avoid some allocation and unnecesssary work in that case.

Reply:
I reworked the code to loop all trivias and check for `JsdocComment::text_is_jsdoc_comment`. This has the benefits

1. We do not create a JsdocComment, (still need a `to_string()` though to forward the reference when using `filter_map`)
2. We can now catch usages in multiple leading comments which brings us some support for `typedefs`